1

lha047 opened the pull request:
Set up project

- Add enonic webpack-starter
- Add Storybook and React for components

Relates to #1

Commit:
#1: init enonic project with webpack starter

Commit:
Merge branch 'master' of https://github.com/ItemConsulting/comments into 1

lha047 committed:
#1: remove LESs loading

lha047 committed:
#1: Update package.json name, version, desc. Remove less dev dep.

lha047 committed:
#1: remove code related to LESS

lha047 committed:
#1: add enonic-ts-codegen as devDep

lha047 committed:
#1: add task to build to generate TS interfaces

lha047 committed:
#1: update app display name, vendor info

cibietici commented on src/main/resources/application.xml:
```diff
@@ -0,0 +1,3 @@
+<application>
+  <description>Edit app description at src/main/resources/application.xml</description>
+</application>
```
Should be fine to make a short description of this application. This description shown underneath the app title on the list of applications installed in XP instance and make life easier for those have to install it and understand what it is for

lha047 replied:
Updated

cibietici commented on src/main/resources/assets/js/example.es6:
```diff
@@ -0,0 +1,4 @@
+export default function example() {
+  const msg = 'Hello from transpiled ES6 client-side code.';
+  console.log(msg);
+}
```
This look like an example file from the starter of Enonic , we might delete it.

cibietici commented on src/main/resources/assets/js/main.es6:
```diff
@@ -0,0 +1,6 @@
+
+import example from './example.es6';
+
+example();
```
This look like also an example file from the starter of Enonic , we might delete it.

cibietici commented on src/main/resources/assets/styles/main.sass:
```diff
@@ -0,0 +1,4 @@
+$text-color: black
+
+.main
+  color: $text-color
```
I suppose we are going to use Less/SCSS rather than sass. But I am not updated with stack we are going to use. In any case we should keep only one of these example files LESS or SASS

lha047 replied:
Deleted

cibietici commented on src/main/resources/assets/styles/main.scss:
```diff
@@ -0,0 +1,5 @@
+
+.main {
+  background-color: $bg-color;
+}
```
See comment from previous file!

cibietici commented on src/main/resources/assets/ts/example.ts:
```diff
@@ -0,0 +1,7 @@
+};
+
+export default example;
+
```
even if we are going to use typescript for this project these file and the next one are examples we are not going to use. Storybook will serve the necessary javascript for the frontend in Enonic's folder anyway. I suggest to remove both files to not confuse

cibietici commented on src/main/resources/assets/ts/main.ts:
```diff
@@ -0,0 +1,6 @@
+
+import example from './example';
+
+example();
```
see previous comment

cibietici commented on src/main/resources/webapp/webapp.ts:
```diff
@@ -0,0 +1,9 @@
+    return {
+        body: thymeleaf.render(view, {})
+    }
+}
```
this is a default example file automatically installed from starter, in case we want to use enonic to make a web app application. I guess we can remove it since we are going not in this scenario

cibietici reviewed (changes requested):
Good start! I commented some files. Mostly is to clean up the project but most important thing is the comment to branch and pull request. we can have a chat about that to establish some main rules for the commit messages.
Another thing we can agree is the task. Not sure what was the task but shouldn't be to install both enonic and storybook?

cibietici commented on src/main/resources/services/check/check.ts:
```diff
@@ -0,0 +1,19 @@
+    contentType: 'application/json',
+    body: JSON.stringify({message: `Params length is ${params.length}.`})
+  }
+};
```
we can remove this as well

cibietici commented on src/main/resources/services/check/check.xml:
```diff
@@ -0,0 +1,5 @@
+  <allow>
+    <principal>role:cms.cm.app</principal>
+  </allow>
+</service>
```
we can remove this as well

lha047 committed:
Removed starter boilerplate

- example
- main
- scss
- webapp

Relates to #1

lha047 committed:
Change app description
- Change app description
- To easier understand what the app does when installed

Relates to #1

lha047 committed:
Change folder structure
- move enonic files to enonic folder
- remove support for JS

Relates to #1

lha047 committed:
Change folder structure
- add client folder
- add storybook react
- to have separate code for client

Relates to #1

lha047 committed:
Setup react, webpack, typescript

- add react and react-dom dep
- add devDep
- add Button test-component
- to code components in React and run project

Relates to #1

cibietici commented on client/tsconfig.json:
```diff
@@ -0,0 +1,71 @@
+    /* Disallow inconsistently-cased references to the same file. */
+    "jsx": "react"
+  }
+}
```
Good practice is not to commit commented lines unless are usefull explanations. I would suggest to remove all lines are not necessary

lha047 replied:
Will remove

cibietici commented on enonic/build.gradle:
```diff
@@ -0,0 +1,106 @@
+    //include "com.enonic.xp:lib-websocket:${xpVersion}"
+    //include "com.enonic.lib:lib-mustache:2.0.0"
+    //include "com.enonic.lib:lib-util:2.0.0"
+}
```
In this case we can keep these commented lines. This file list all libraries Enonic ships with and according to how project develop we might need or not these packages.

cibietici commented on enonic/src/main/resources/template/main.html:
```diff
@@ -0,0 +1,15 @@
+<body>
+    Hello from Comments!
+</body>
+</html>
```
This is also a file automatically created by the enonic starter and we are not going to use it, if we are going to use this template we need to move the bundle script including at the end of the body. I suggest to remove it since we are going to write our own pages and parts in enonic.

cibietici reviewed (changes requested):
Excellent! we are getting closer. I suggest just few cleaning to files and structure

lha047 committed:
Clean up client project

- use vanilla TS
- to clean up project and make it easier to reuse for other projects

Relates to #1

cibietici commented on client/package.json:
```diff
@@ -0,0 +1,47 @@
+  "dependencies": {},
+  "devDependencies": {
+    "@babel/core": "^7.11.1",
+    "@fullhuman/postcss-purgecss": "^2.3.0",
```
Since we agreed to go for vanila js/ts and not any stylesheet frameworks  we can also remove this purgecss dependencies.

cibietici commented on client/webpack.config.js:
```diff
@@ -0,0 +1,161 @@
+        'my-2', 'my-4', 'my-8', 'my-12', 'my-16', 'my-20', 'my-24',
+    ],
+    defaultExtractor: content => content.match(/[\w-/:]+(?<!:)/g) || []
+});
```
since we are going to remove purgecss dependency we can remove const from line 7 to 48

cibietici reviewed (changes requested):
Well done! just last two things to remove related to purgecss dependency, which we are not going to use, and we are ready to merge to master.

lha047 committed:
Remove purgecss

- To claan up project for only custom css

Relates to # 1
